iOS/Flows/People/PeopleCoordinator.swift

```diff
@@ -36,9 +43,230 @@ class PeopleCoordinator: PresentableCoordinator<[Person]> {
+            return await self.presentConnectionFlow(for: user)
+        } else if let reservation = self.getReservation(for: contact) {
+            // Get a valid reservation object that we can use to invite this person. Then prompt the user
+            // to send them a text
+            await self.sendText(to: contact, with: reservation)
+        }
+
+        return nil
+    }
+
+    /// Returns the reservation that should be used to invite this person. Nil is returned if there are no valid reservation objects left that can be used.
+    private func getReservation(for contact: CNContact) -> Reservation? {
+        let contactId = contact.identifier
+
+        // First check to see if there's already a reservation object associated with this contact.
+        if let existingReservation = self.reservations.first(where: { reservation in
+            return reservation.contactId == contactId
+        }) {
+            return existingReservation
+        }
+
+        // If there are no reservations associated with the contact, then return an unused reservation.
+        return self.reservations.first { reservation in
+            return reservation.contactId == nil
+        }
+    }
+
+    // MARK: - Existing Users Flow
+
+    /// Returns a Jibber user that has the same phone number as the passed in contact.
+    private func findUser(with contact: CNContact) async -> User? {
+        // Search for user with phone number
+        guard let phone = contact.findBestPhoneNumber().phone?.stringValue.removeAllNonNumbers() else {
+            return nil
+        }
+
+        return try? await User.getFirstObject(where: "phoneNumber", contains: phone)
+    }
+
+    /// Presents an alert that asks if the user wants to connect with the passed in user. Finishes once a connection is made or the user cancels.
+    func presentConnectionFlow(for user: User) async -> Connection? {
+        return await withCheckedContinuation { continuation in
+            let title = LocalizedString(id: "", arguments: [user.fullName], default: "Connect with @(name)?")
+            let titleText = localized(title)
+
+            let body = LocalizedString(id: "", arguments: [user.fullName], default: "@(name) has an account. Tap OK to send the request. (This will NOT consume one of your RSVPs)")
+            let bodyText = localized(body)
+            let alert = UIAlertController(title: titleText,
+                                          message: bodyText,
+                                          preferredStyle: .alert)
+
+            let cancel = UIAlertAction(title: "Cancel", style: .cancel) { (_) in
+                continuation.resume(returning: nil)
+            }
+
+            let ok = UIAlertAction(title: "Ok", style: .default) { [unowned self] (_) in
+                Task {
+                    let connection = await self.createConnection(with: user)
+                    continuation.resume(returning: connection)
```

Comment:
The presentation is wrapped in a continuation block, and we resume when the alert buttons are tapped and a connection is made.